arches_lingo/src/arches_lingo/components/scheme/report/SchemeLabel.vue

```diff
@@ -1,54 +1,108 @@
 <script setup lang="ts">
 import { useGettext } from "vue3-gettext";
+import { onMounted, ref } from "vue";
+import { useRoute } from "vue-router";
+
+import { VIEW, EDIT, OPEN_EDITOR } from "@/arches_lingo/constants.ts";
+import type {
+    DataComponentMode,
+    SchemeInstance,
+} from "@/arches_lingo/types.ts";
+import { deleteSchemeLabelTile, fetchSchemeLabel } from "@/arches_lingo/api.ts";
 import SchemeReportSection from "@/arches_lingo/components/scheme/report/SchemeSection.vue";
-import { ref } from "vue";
-import DataTable from "primevue/datatable";
-import Column from "primevue/column";
-const labels = ref([
-    {
-        code: "1",
-        name: "label 1",
-        category: "category 1",
-        quantity: "1",
-    },
-    {
-        code: "2",
-        name: "label 2",
-        category: "category 2",
-        quantity: "2",
-    },
+import LabelViewer from "@/arches_lingo/components/generic/LabelViewer.vue";
+import { useToast } from "primevue/usetoast";
+
+const schemeInstance = ref<SchemeInstance>();
+const { $gettext } = useGettext();
+const toast = useToast();
+const route = useRoute();
+
+withDefaults(
+    defineProps<{
+        mode?: DataComponentMode;
+        args?: Array<object>;
+    }>(),
     {
-        code: "3",
-        name: "label 3",
-        category: "category 3",
-        quantity: "3",
+        mode: VIEW,
+        args: () => [],
     },
-]);
-const { $gettext } = useGettext();
+);
+
+defineExpose({ save, getSectionValue });
+
+const emits = defineEmits([OPEN_EDITOR]);
+
+onMounted(async () => {
+    getSectionValue();
+});
+
+async function getSectionValue() {
+    const result = await fetchSchemeLabel(route.params.id as string);
+    schemeInstance.value = {
+        appellative_status: result.appellative_status,
+    };
+}
+
+async function deleteSectionValue(tileId: string) {
+    const result = await deleteSchemeLabelTile(tileId);
+    if (result) {
```

Comment:
This will never happen per the API call - it will return true or throw.  The only case it would throw is that the delete failed - in which case something exceptional has happend (has the tile already been deleted?).  Should we catch that?  Should we return false instead of throwing?  Catch and toast?

Reply:
Yeah, there should be error handling for sure. IMO wrap this in a try/catch that will raise a Toast error. I'd say same with `getSectionValue` and any other API.